tests/tools/oecertdump/enc/enc.cpp

```diff
@@ -0,0 +1,310 @@
+
+    result = OE_OK;
+done:
+    return result;
+}
+
+// input: input_data and input_data_len
+// output: key, key_size
+oe_result_t generate_key_pair(
+    int key_type,
+    uint8_t** public_key,
+    size_t* public_key_size,
+    uint8_t** private_key,
+    size_t* private_key_size)
+{
+    oe_result_t result = OE_FAILURE;
+    oe_asymmetric_key_params_t params;
+    char user_data[] = "test user data!";
+    size_t user_data_size = sizeof(user_data) - 1;
+    uint8_t* local_public_key = NULL;
+    uint8_t* local_private_key = NULL;
+
+    if (key_type == MBEDTLS_PK_ECKEY)
+    {
+        params.type =
+            OE_ASYMMETRIC_KEY_EC_SECP256P1; // MBEDTLS_ECP_DP_SECP256R1
+        params.format = OE_ASYMMETRIC_KEY_PEM;
+        params.user_data = user_data;
+        params.user_data_size = user_data_size;
+        result = oe_get_public_key_by_policy(
+            OE_SEAL_POLICY_UNIQUE,
+            &params,
+            public_key,
+            public_key_size,
+            NULL,
+            NULL);
+        OE_CHECK(result);
+
+        result = oe_get_private_key_by_policy(
+            OE_SEAL_POLICY_UNIQUE,
+            &params,
+            private_key,
+            private_key_size,
+            NULL,
+            NULL);
+        OE_CHECK(result);
+    }
+    else if (key_type == MBEDTLS_PK_RSA)
+    {
+        int res = -1;
+        mbedtls_ctr_drbg_context ctr_drbg_context;
+        mbedtls_entropy_context entropy_context;
+        mbedtls_pk_context pk_context;
+        size_t local_public_key_size = 512;
+        size_t local_private_key_size = 2048;
+
+        mbedtls_ctr_drbg_init(&ctr_drbg_context);
+        mbedtls_entropy_init(&entropy_context);
+        mbedtls_pk_init(&pk_context);
+
+        // Initialize entropy.
+        res = mbedtls_ctr_drbg_seed(
+            &ctr_drbg_context, mbedtls_entropy_func, &entropy_context, NULL, 0);
+        if (res != 0)
+        {
+            OE_TRACE_ERROR("mbedtls_ctr_drbg_seed failed.");
+            goto clean_rsa;
+        }
+
+        // Initialize RSA context.
+        res = mbedtls_pk_setup(
+            &pk_context, mbedtls_pk_info_from_type(MBEDTLS_PK_RSA));
+        if (res != 0)
+        {
+            OE_TRACE_ERROR("mbedtls_pk_setup failed (%d).", res);
+            goto clean_rsa;
+        }
+
+        // Generate an ephemeral 2048-bit RSA key pair with
+        // exponent 65537 for the enclave.
+        res = mbedtls_rsa_gen_key(
+            mbedtls_pk_rsa(pk_context),
+            mbedtls_ctr_drbg_random,
+            &ctr_drbg_context,
+            2048,
+            65537);
+        if (res != 0)
+        {
+            OE_TRACE_ERROR("mbedtls_rsa_gen_key failed (%d)\n", res);
+            goto clean_rsa;
+        }
+
+        /* Call again with the allocated memory. */
+        local_public_key = (uint8_t*)malloc(local_public_key_size);
+        if (local_public_key == NULL)
+            OE_RAISE(OE_OUT_OF_MEMORY);
```

Comment:
here and in 151, if malloc fails, it will go to done instead of to clean_rsa, and will leak the mbedtls memory.  This looks like a bug, hence -1 for this bug.